src/Dapr.Extensions.Configuration/DaprSecretStoreConfigurationSource.cs

```diff
@@ -29,22 +29,22 @@ public class DaprSecretStoreConfigurationSource : IConfigurationSource
         /// <summary>
         /// Gets or sets the custom key delimiters. Contains the '__' delimiter by default.
         /// </summary>
-        public IList<string> KeyDelimiters { get; set; } = new List<string> { "__" };
+        public IList<string>? KeyDelimiters { get; set; } = new List<string> { "__" };
 
         /// <summary>
         /// Gets or sets the secret descriptors.
         /// </summary>
-        public IEnumerable<DaprSecretDescriptor> SecretDescriptors { get; set; }
+        public IEnumerable<DaprSecretDescriptor>? SecretDescriptors { get; set; }
```

Comment:
Does it work to set this to `{ get; set; } = Array.Empty<DaprSecretDescriptors>()`

Reply:
Looks like I have the answer - this should be nullable after all. https://github.com/dapr/dotnet-sdk/pull/636/files#diff-e03608cfa6c16ee7656cce2f6bbe55da1dea01c3994fd6822a7adb20fa44d8cfL139

Unrelated to your PR we made a mistake here... it's not good to have a semantic meaning between *null collection* and *empty collection*.